proto/teleport/quicpeering/v1alpha/dial.proto

```diff
@@ -0,0 +1,57 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package teleport.quicpeering.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/teleport/quicpeering/v1alpha;quicpeeringv1alpha";
+
+// Sent from the client to the server in a fresh QUIC stream to dial a server
+// through a QUIC proxy peering connection. The message is sent in protobuf
+// binary format, prefixed by its length encoded as a little endian uint32.
+message DialRequest {
+  string server_id = 1;
+  string conn_type = 2;
+
+  Addr source = 3;
+  Addr destination = 4;
+
+  // The time of the client, must be provided and within 5 minutes of the local
+  // server time for 0-RTT requests.
+  google.protobuf.Timestamp timestamp = 5;
+  // A random id for each dial request, must be provided and unique among
+  // requests presented in a 10-20 minute window for 0-RTT requests.
+  fixed64 nonce = 6;
+}
+
+// A stringy Go net.Addr.
+message Addr {
+  string network = 1;
+  string addr = 2;
+}
+
+// Sent from the server to the client as a response to a DialRequest. The
+// message is likewise sent in protobuf binary format, prefixed by its length
+// encoded as a little endian uint32.
```

Comment:
nit: conventionally data sent over the wire is big endian.  GRPC performs length prefixing using big endian uint32s.  unless there is some compelling reason to not to, I'd suggest sticking with that convention.

Reply:
Counterpoint: it's 2024.

gRPC over HTTP/2 uses big endian for length prefixes because the HTTP/2 spec uses big endian and that's just how they happened to write the spec; protobuf itself uses little endian for any fixed-length integers, so "convention" should clearly not be a factor in any new protocol.

Reply:
Alright, ignoring convention...  little endian byte order is an affront to god and nature and has no place in a civilized codebase.  Especially for the case of a home-brewed API that we might be called upon to debug at some point, since visually parsing little endian data is annoying/weird.

Reply:
I also have a slight preference for big endian for over-the-wire data, if nothing else because I would expect it to be the case. That said, as long as it's well documented I'm OK with it.